dali/util/s3_filesystem.h

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_UTIL_S3_FILESYSTEM_H_
+#define DALI_UTIL_S3_FILESYSTEM_H_
+
+#include <aws/core/Aws.h>
+#include <aws/s3/S3Client.h>
+#include <cstdio>
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+#include "dali/core/common.h"
```

Comment:
That's a heavy hammer - wouldn't api_helper suffice?
```suggestion
#include "dali/core/api_helper.h"
```